packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -106,6 +108,9 @@ export class SummarizerNode implements IRootSummarizerNode {
         try {
             const result = await this.summarizeInternalFn(fullTree);
             this.wipLocalPaths = { localPath: EscapedPath.create(result.id) };
+            if (result.pathPartsForChildren !== undefined) {
```

Comment:
Why does the summarize result need to have "pathPartsForChildren"? Can't it get the information from the result itself similar to what parseSummaryForSubtrees does?

Reply:
Also, can you add a comment here explaining what this code is doing.

Reply:
It could try to do it the same way `parseSummaryForSubtrees` does, but ideally we wouldn't have `parseSummaryForSubtrees` at all, since it is really a ContainerRuntime detail leaking into SummarizerNode. I am trying to reduce the number of places where we use `parseSummaryForSubtrees` so that it is easier to pull that logic out in the future. It isn't too bad during `loadBaseSummary`, but it gets trickier with `refreshLatestSummaryAck`.